src/PKSim.Core/Services/SimulationModelCreator.cs

```diff
@@ -101,62 +101,108 @@ private void hideAllUndefinedParameterForMolecules(Simulation simulation)
-            var relExp = expressionContainer.RelativeExpressionParameter;
-
             if (expressionContainer.IsSurrogate())
             {
-               string relExpName;
-               if (expressionContainer.IsBloodCell())
-                  relExpName = CoreConstants.Parameter.RelExpBloodCell;
-               else if (expressionContainer.IsPlasma())
-                  relExpName = CoreConstants.Parameter.RelExpPlasma;
-               else if (expressionContainer.IsVascularEndothelium())
-                  relExpName = CoreConstants.Parameter.RelExpVascEndo;
-               else
-                  continue;
-
-               string relExpNormName = CoreConstants.Parameter.NormParameterFor(relExpName);
-               updateFromIndividualParameter(globalMoleculeContainer.Parameter(relExpName), relExp, individual, molecule);
-               updateFromIndividualParameter(globalMoleculeContainer.Parameter(relExpNormName), relExpNorm, individual, molecule);
+               updateReferenceToIndividualParametersForSurrogateContainer(individual, molecule, expressionContainer, globalMoleculeContainer);
             }
             else
             {
-               //not a global parameter simply update the norm  parameters
-               var allContainers = _expressionContainersRetriever.AllContainersFor(individual.Organism, allOrganismContainers, molecule, expressionContainer);
-               foreach (var container in allContainers)
-               {
-                  var amount = moleculeAmountPath[_entityPathResolver.ObjectPathFor(container).AndAdd(molecule.Name).ToString()];
-                  if (amount == null) continue;
-
-                  updateFromIndividualParameter(amount.Parameter(CoreConstants.Parameter.RelExpNorm), relExpNorm, individual, molecule);
-                  updateFromIndividualParameter(amount.Parameter(CoreConstants.Parameter.RelExp), relExp, individual, molecule);
-               }
+               updateReferenceToIndividualParametersForStandardContainer(individual, molecule, expressionContainer, allOrganismContainers, moleculeAmountPath);
             }
          }
       }
 
+      private void hideInterstialParametersForIntracellularLocalizationInTissue(Individual individual, IndividualMolecule molecule, IList<IContainer> allOrganismContainers, PathCache<IMoleculeAmount> moleculeAmountPath)
+      {
+         var protein = molecule as IndividualProtein;
+
+         if (protein?.TissueLocation != TissueLocation.Intracellular || protein.IntracellularVascularEndoLocation != IntracellularVascularEndoLocation.Interstitial)
+            return;
+
+         var vascularEndothelium = protein.ExpressionContainer(CoreConstants.Compartment.VascularEndothelium);
+         if (vascularEndothelium == null)
+            return;
+
+         var allInterstitialContainer = _expressionContainersRetriever.AllContainersFor(individual.Organism, allOrganismContainers, protein, vascularEndothelium);
```

Comment:
This is correct. That being said, if we ever changed which container is affected by the molecule  properties, we won't have to change that code hopefully (the knowledge of what container are depending on vascular endothelium is only in one place)